### ISSUE TITLE
Export Web Page - number of unread titles does not match OpenAudible nor Audible - How do I Open?

### ISSUE THREAD

**Issue by lyallp** [openaudible/openaudible#1474]
Version 4.4.7, on Windows 10.
Syncrhonised with Audible (USA)  successfully after purchasing 22 books. (just started a new year!)

In OpenAudible, I added Status column to the display grid
Sorted by status
OpenAudible Application in agreement with Audible about book read status, 28 books as "Unread"

Exported Web Page
Customised web page to include a 'Status' column which shows the book status.
Sorted by Status
Only 8 books are showing as "Unread", two as "Reading" and the rest as "Finished".
Repeated export again, just in case. Same result.

**Comment by lyallp** [comment]
FYI: Books.js in the web page directory only contains 8 'Unread' entries. (I searched for **"read_status":"Unread"** - including quotes, case insensitive)

So, the web page is showing what is in the Books.js
The Books.json in the OpenAudible storage folder has the correct number of 'Unread' books.
So, somewhere in translation from Books.json to web:Books.js, rows appear to be going missing.

**Comment by openaudible** [comment]
The read_status should match what we show in the book table. You can turn that on by right clicking the table, going to Table Settings, and checking the read_status checkbox.
If you look at the web's books.json file it should have entries for read_status, such has `  "read_status": "Unread",`

The web viewer doesn't currently show the read status in the web table, and I don't believe you can type in "Unread" or "Finished" in the search box -- but that might be something that can be added as a special case search. 

Make sure you're looking at the correct web output directory.. And if you wanted to create a few screenshots, you could send them to info at openaudible dot org.

**Comment by lyallp** [comment]
I modified the web viewer to show read status, this is why I reported the discrepancy between OpenAudible application and the exported Web page.

**Comment by openaudible** [comment]
You'll need to debug this:

```
// OpenAudible.js web page to display audio book library.

// filter out (hide) books not in filter text. If no filter text, all books are shown.
function filter() {
    const text = $("#filter").val();
    const rex = new RegExp(text, 'i');
    $('.searchable tr').hide();
    $('.searchable tr').filter(function () {
        const row = $(this).text();
        return rex.test(row);
    }).show();
}
```
I made a little change to add the const row which should be all of the text in one row of book data.
Maybe set a breakpoint on it and make sure you see the "unread" as part of the text of row.

But we definitely fill in the read_status every time you export to web. It should match, and if it doesn't, check the modification date of the books.json files.. make sure you're looking at the same file that is being exported.

**Comment by lyallp** [comment]
I have confirmed my suspicions.
in the openaudible storage directory is a file called books.json
perform (in linux : jc is the json conversion tool) 
$ cat books.json | jc --indent 1 | grep -i unread

Then, I went into the Web export folder, manually edited the books,js to remove the "window.myBooks=" from the start of the file and saved it to temp.js
I then 
$ cat temp.js | jc --indent 1  |  grep -i unread 

The number of rows was 6 in my export to the web, and 22 in the books.js database used by OpenAudible.
I have just updated to the latest release of OpenAudible 4.4.8 on Windows 11 and re-exported my web page.
